clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -168,6 +171,8 @@ private boolean isTransitionValid(State source, State target) {
                     return source == IN_TRANSACTION || source == ABORTABLE_ERROR;
                 case ABORTABLE_ERROR:
                     return source == IN_TRANSACTION || source == COMMITTING_TRANSACTION || source == ABORTABLE_ERROR;
+                case FATAL_BUMPABLE_ERROR:
+                    return source != FATAL_ERROR;
```

Comment:
It might make sense to have more fine tuned restriction when we can get into FATAL_BUMPABLE_ERROR. For example we might not want to get there from UNINITIALIZED

Reply:
Yes, good catch, cleared that one up